legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java

```diff
@@ -118,12 +121,20 @@ public String getName() {
         return "sql_action";
     }
 
+    /**
+     * Prepare and execute rest SQL request. In the event the V2 SQL engine fails, the V1
+     * engine attempts the query.
+     * @param request : Rest request being made.
+     * @param client : Rest client for making the request.
+     * @return : Resulting values for request.
+     */
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
         Metrics.getInstance().getNumericalMetric(MetricName.REQ_TOTAL).increment();
         Metrics.getInstance().getNumericalMetric(MetricName.REQ_COUNT_TOTAL).increment();
 
         LogUtils.addRequestId();
+        newSqlQueryHandler.setErrorStr("");
```

Comment:
Is it necessary?

Reply:
Since this class is a singleton we can potentially re-use old error messages. This ensures we are starting with an empty error message.

Reply:
if it's singleton will errorStr handle concurrent errors?